doc/md/fundamentals/3-declarations/1-variable-declarations.md

```diff
@@ -0,0 +1,51 @@
+---
+sidebar_position: 1
+---
+
+# Variable declarations
+
+In Motoko variables represent state within actors, classes, and modules. Variables are declared using either `let` (immutable) or `var` (mutable).
+
+| Feature                 | `let`    | `var`       |
+|-------------------------|---------|-------------|
+| Reassignment           | Not allowed | Allowed |
+| Value modification     | Cannot be updated after assignment. | Can be updated after assignment. |
+
+:::info
+
+In `persistent` actors, `let` and `var` bindings are treated as stable by default. Actor state must use [stable types](https://internetcomputer.org/docs/motoko/fundamentals/types/stable-types), otherwise, declarations must be marked `transient`.
+
+The `persistent` keyword eliminates the need for explicit `stable` annotations, and `transient` replaces the legacy `flexible` modifier.
+
+Motoko encourages a programming style that favours immutability (using `let` over `var`), reflecting functional programming principles.
+:::
+
+## Immutable variables
+
+A `let` declaration names the value of an expression so that the expression can be evaluated once and its result, a value, referenced by that name. The compiler ensures that any attempt to modify an immutable variable results in an error.
+
+```motoko no-repl
+let x = 10;
+x := 20; // Error: Cannot assign to immutable variable
+```
+
```

Comment:
It might be better do say let bindings support pattern matching here, with an example
````suggestion

The left hand side of a `let` can be also be a more general pattern, naming the components of a value by matching its structure:

For example, the declaration 

``` motoko
let (fst, snd) = (1, 2); 
```
uses the pattern `(fst, snd)` to name the components of the pair `(1,2)`. The value of `fst` is `1` and the value of `snd` is 2. Both `fst` and `snd` are immutable.
````

And then later say that `var` doesn't not support pattern matching.